tempesta_fw/sock.c

```diff
@@ -198,13 +198,9 @@ ss_tcp_process_proto_skb(struct sock *sk, unsigned char *data, size_t len,
 
 	read_lock(&sk->sk_callback_lock);
 	r = SS_CALL(connection_recv, sk, data, len);
-	if (r == SS_POSTPONE) {
-		SS_CALL(postpone_skb, sk->sk_user_data, skb);
-		r = SS_OK;
-	}
 	read_unlock(&sk->sk_callback_lock);
 
-	return r;
+	return (r == SS_POSTPONE) ? SS_OK : r;
```

Comment:
Dirty fix.

If we don't want to process TFW_POSTPONE return code, the all called functions (tfw_http_req_process() and tfw_http_resp_process()) should not return the value. However, the functions do handles POSTPONE logic, so they should leave the place where the logic is handled while the lower, network, layer should be unaware about POSTPONE.